apps/system/js/nfc_handover_manager.js

```diff
@@ -14,689 +13,716 @@
-    }
+    },
 
-    window.addEventListener('bluetooth-adapter-added', function() {
-      self._debug('bluetooth-adapter-added');
-      var req = self.bluetooth.getDefaultAdapter();
-      req.onsuccess = function bt_getAdapterSuccess() {
-        self.settingsNotified = false;
-        self.defaultAdapter = req.result;
-        self._debug('MAC address: ' + self.defaultAdapter.address);
-        self._debug('MAC name: ' + self.defaultAdapter.name);
-        /*
-         * Call all actions that have queued up while Bluetooth
-         * was turned on.
-         */
-        for (var i = 0; i < self.actionQueue.length; i++) {
-          var action = self.actionQueue[i];
-          action.callback.apply(self, action.args);
+    _hasInititialized: false,
+
+    /**
+     * Logs message in logcat
+     * @param {String} msg debug messages
+     * @param {Object} opObject object to printed after doing JSON.stringfy
+     * @memberof NfcHandoverManager.prototype
+     */
+    _logVisibly: function _logVisibly(msg, optObject) {
+        var output = '[NfcHandoverManager]: ' + msg;
+        if (optObject) {
+          output += JSON.stringify(optObject);
         }
-        self.actionQueue = [];
-      };
-      req.onerror = function bt_getAdapterError() {
-        self._logVisibly('event listner: Failed to get bluetooth adater');
-      };
-    });
-
-    window.addEventListener('bluetooth-disabled', function() {
-      self._debug('bluetooth-disabled');
-      self._clearBluetoothStatus();
-    });
-
-    window.navigator.mozSetMessageHandler('nfc-manager-send-file',
-      function(msg) {
-        self._debug('In New event nfc-manager-send-file' + JSON.stringify(msg));
-        self.handleFileTransfer(msg.peer, msg.blob, msg.requestId);
-    });
-
-    SettingsListener.observe('nfc.debugging.enabled', false,
-                             (enabled) => { this.DEBUG = enabled; });
-
-    window.addEventListener('nfc-transfer-started',
-      this._transferStarted.bind(this)
-    );
-  },
-
-  /**
-   * Save the on/off status of Bluetooth.
-   * @memberof NfcHandoverManager.prototype
-   */
-  _saveBluetoothStatus: function _saveBluetoothStatus() {
-    if (!this.bluetoothStatusSaved) {
-      this.bluetoothStatusSaved = true;
-      this.bluetoothAutoEnabled = !this.bluetooth.enabled;
-    }
-  },
-
-  /**
-   * Restore the Bluetooth status.
-   * @memberof NfcHandoverManager.prototype
-   */
-  _restoreBluetoothStatus: function _restoreBluetoothStatus() {
-    if (!this.isHandoverInProgress() &&
-        BluetoothTransfer.isSendFileQueueEmpty) {
-      if (this.bluetoothAutoEnabled) {
-        this._debug('Disabling Bluetooth');
-        this.settings.createLock().set({'bluetooth.enabled': false});
-        this.bluetoothAutoEnabled = false;
+        console.log(output);
+    },
+
+    /**
+     * Initializes event and message handlers, initializes properties.
+     * @memberof NfcHandoverManager.prototype
+     */
+    start: function start() {
```

Comment:
I worry about nfc_handover_manager has not stop function yet (which i will implement later). However, it seems to be ok  because module would not start again if no stop is called.